contributing-device-support.md

```diff
@@ -1,120 +1,207 @@
 
-    - samples/bblayers.conf.sample file in which all the required yocto layers are listed (see this [bblayers.conf.sample][bblayers.conf.sample intel] for an example)
+- `recipes-core/images` directory, which contains at least a `resin-image.bbappend` file. Depending on the type of board you are adding support for, you should have your device support either just `resin-image` or both `resin-image-flasher` and `resin-image`. Generally, `resin-image` is for boards that boot directly
+from external storage (these boards do not have internal storage to install resin.io on). `resin-image-flasher` is used when the targeted board has internal storage so this flasher image is burned onto an SD card or USB stick that is used for the initial boot. When booted, this flasher image will automatically install resin.io on internal storage.
 
-    - samples/local.conf.sample file which defines part of the build configuration (see the meta-resin [README.md][meta-resin-readme] for an overview of some of the variables used
-    in the local.conf.sample file). An existing file can be used (e.g. [local.conf.sample][local.conf.sample intel]) but making sure the "Supported machines" area lists the appropriate machines this repository is used for.
+  The `resin-image.bbappend` file shall define the following variables:
 
-- recipes-containers/docker-disk directory - contains docker-resin-supervisor-disk.bbappend that shall define the following variable(s):
+    - `IMAGE_FSTYPES_<yocto-machine-name>`: this variable is used to declare the type of the produced image (it can be ext3, ext4, resin-sdcard etc. The usual type for a board that can boot from SD card, USB, is "resin-sdcard").
 
-    - SUPERVISOR_REPOSITORY_`<yocto-machine-name>`
+    - `RESIN_BOOT_PARTITION_FILES_<yocto-machine-name>`: this allows adding files from the build's deploy directory into the vfat formatted resin-boot partition (can be used to add bootloader config files, first stage bootloader, initramfs or anything else needed for the booting process to take place for your particular board). If the board uses different bootloader configuration files when booting from either external media (USB thumb drive, SD card etc.) or from internal media (mSATA, eMMC etc) then you would want make use of this variable to make sure the different bootloader configuration files get copied over and further manipulated as needed (see `INTERNAL_DEVICE_BOOTLOADER_CONFIG_<yocto-machine-name>` and `INTERNAL_DEVICE_BOOTLOADER_CONFIG_PATH_<yocto-machine-name>` below). Please note that you only reference these files here, it is the responsibility of a `.bb` or `.bbappend` to provide and deploy them (for bootloader config files this is done with an append typically in `recipes-bsp/<your board's bootloader>/<your board's bootloader>.bbappend`, see [resin-intel grub bbappend][resin-intel grub append] for an example)
```

Comment:
"then you would want make use of this variable" should be probably:

"then you would want to make use of this variable"